### DISCUSSION TITLE
Ensuring that App is closed before trying to remove log file?

### DISCUSSION THREAD

**Discussion by mr-intj** [wixtoolset/issues#6415] [Questions] [answered]
For several reasons, I'd like to ensure that our app isn't running (or it is closed automatically, or the user is prompted to close it) during a major upgrade or uninstall.

When uninstalling or updating our app, the user currently gets an Access Denied message box if the app is still running. The message names a log file that our app is holding open. A prompt appears letting the user that the file is in use, but since it's a log file that users aren't normally aware of, our users may not realize that they need to close one of our apps to procede.

This question is primarily about ensuring that the app isn't running prior to attempting the removal of any log files. Assuming that I use something like `<util:CloseApplication ...>`, I would need to _**ensure**_ that this took place before any file deletions took place.

I originally thought that I could ensure basic ordering using nesting of components and component groups, but I realize now that control is very limited (or I'm missing something).

When I search for possible solutions, everyone's go-to seems to be a custom action, but the blogs and guides I've read strongly recommend against them.

Do I just need to ensure that the CloseApplication occurs prior to the processing of the items in the RemoveFile table?

Is there a generally-agreed-upon best practice or range of solutions for this?

Thanks.

**Comment by BMurri** [comment, marked as answer]
`<util:CloseApplication>` runs before `<RemoveFile>` by default, so that should be sufficient.
